### PR TITLE
#169597894 Preventing direct .env calls

### PR DESCRIPTION
**Pull request by Mastel22**
#### What does this PR do?
Create a single config module that will handle all of the process.env calls in one maintainable place.
#### Description of Task to be completed?
- Added the config module to handle the dotenv calls
- Edited the current files that made those direct calls
#### How should this be manually tested?
-N/A
#### Any background context you want to provide?
-N/A
#### What are the relevant pivotal tracker stories?
[#169597894](https://www.pivotaltracker.com/story/show/169597894)
#### Questions:
None

### CODE REVIEW

File: src/models/index.js [hunk 1]
```diff
@@ -2,13 +2,12 @@
 import { readdirSync } from 'fs';
 import { basename as _basename, join } from 'path';
 import Sequelize from 'sequelize';
+import configEnv from '../config';
 
 const basename = _basename(__filename);
-const env = process.env.NODE_ENV || 'development';
 // eslint-disable-next-line import/no-dynamic-require
```
Comment (timolinn) [inline src/models/index.js]: Great work here! We don't need to this line anymore.

**Review by nshaphan — DISMISSED** [review]
LGTM

File: src/index.js [hunk 2]
```diff
@@ -7,9 +7,14 @@ import errorhandler from 'errorhandler';
 import router from './routes/index';
 import ErrorHandler from './utils/error';
 import logger from './utils/winston';
+import config from './config';
 
 const isDevelopment = process.env.NODE_ENV;
```
Comment (RaymondMwaura) [inline src/index.js]: This line calls '.env' directly. Maybe you can use the config object you've already imported.

File: src/models/index.js [hunk 3]
```diff
@@ -2,13 +2,12 @@
...
-const config = require(join(__dirname, '/../database/config.js'))[env];
+const config = configEnv.database;
 const db = {};
-
 let sequelize;
 if (config.use_env_variable) {
   sequelize = new Sequelize(process.env[config.use_env_variable]);
```
Comment (RaymondMwaura) [inline src/models/index.js:12]: This line calls '.env' directly. Maybe you can use the config object you've already imported.

**Review by RaymondMwaura — CHANGES_REQUESTED** [review]
I think this is a job well done. I have left some comments for you to look at. Also, the migration scripts ('migrate' and 'migrate:undo')don't seem to be working anymore. Maybe you can liaise with @b0nbon1 to verify this. I also think we can remove the 'dotenv' call that's required in both the 'start' and 'watch' scripts in 'package.json'.

File: src/config/env/default.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  secret: process.env.SECRET,
+  database: {
+    url: process.env.DATABASE_URL,
+    dialect: 'postgres',
+  },
+  debug: false,
+};
+
+// Set the current environment or default to 'development'
+process.env.NODE_ENV = process.env.NODE_ENV || 'development';
```
Comment (timolinn) [inline src/config/env/default.js]: I don't think this line is necessary, why not just do?
```javascript
    config.env = process.env.NODE_ENV || 'development';
```

File: src/models/index.js (hunk 3 above)
Comment (Mastel22) [reply src/models/index.js:12]: That's not the projects env it's sequelize's

File: src/index.js (hunk 2 above)
Comment (Mastel22) [reply src/index.js]: Oh yes.Corrected now

File: src/models/index.js (hunk 1 above)
Comment (Mastel22) [reply src/models/index.js]: Oh yes.Corrected now

File: config/config.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "operatorsAliases": false
+  },
+  "production": {
+    "username": "root",
+    "password": null,
+    "database": "database_production",
+    "host": "127.0.0.1",
+    "dialect": "mysql",
+    "operatorsAliases": false
+  }
```
Comment (nshaphan) [inline config/config.json]: I think this needs to use environment variables from respective configuration file. They do not have to be hardcoded

**Review by RaymondMwaura — DISMISSED** [review]
This is now okay.

**Review by RaymondMwaura — DISMISSED** [review]
This is now complete.

**Review by nshaphan — DISMISSED** [review]

File: src/config/env/default.js (hunk 4 above)
Comment (Mastel22) [reply src/config/env/default.js]: Oh yes.It's done.

**Review by RaymondMwaura — DISMISSED** [review]
This looks okay.

**Review by nshaphan — DISMISSED** [review]
LGTM

**Comment by kipropbrian** [comment]
You have 2 commits with similar names, and remove the `:` in both your PR titles.

**Review by nshaphan — APPROVED** [review]

**Review by timolinn — APPROVED** [review]
Great work Mastel. It all looks good to me.

**Review by RaymondMwaura — APPROVED** [review]
Great job!

**Pull request merged**